sklearn/decomposition/_kernel_pca.py

```diff
@@ -69,28 +71,28 @@ class KernelPCA(TransformerMixin, BaseEstimator):
         Maximum number of iterations for arpack.
         If None, optimal value will be chosen by arpack.
 
-    remove_zero_eig : boolean, default=False
+    remove_zero_eig : bool, default=False
         If True, then all components with zero eigenvalues are removed, so
         that the number of components in the output may be < n_components
         (and sometimes even zero due to numerical instability).
         When n_components is None, this parameter is ignored and components
         with zero eigenvalues are removed regardless.
 
-    random_state : int, RandomState instance, default=None
+    random_state : int, RandomState instance or None, default=None
```

Comment:
```suggestion
    random_state : int, RandomState instance, default=None
```
when `default=None` we don't put it inside the type

Reply:
We should do it for random_state because `None` has a very specific meaning and doesn't correspond to the usual meaning of `None` as e.g. in `sample_weight=None`

Reply:
this is the same as for  `n_jobs`